CSS Server/Models/CameraManager.cs

```diff
@@ -38,5 +41,57 @@ public List<Camera> Cameras
             //set { myVar = value; }
         }
 
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="webSocket"></param>
+        /// <returns></returns>
+        /// <exception cref="NotImplementedException"></exception>
+        public async Task<Camera> ValidateCameraConnection(WebSocket webSocket)
+        {
+            //Read the first message of the connection for validation purposes.
+            byte[] buffer = new byte[1024 * 4];
+            WebSocketReceiveResult firstResult = await webSocket.ReceiveAsync(new ArraySegment<byte>(buffer), CancellationToken.None);
+
+            JObject firstData = JObject.Parse(Encoding.UTF8.GetString(buffer, 0, buffer.Length));
+
+            bool isIncorrectRequest = false;
+
+            //Try to parse the password value from the sent json.
+            JToken jPassword;
+            if(!firstData.TryGetValue("password", out jPassword))
+                isIncorrectRequest = true;
+
+            //Try to parse the id value from the sent json.
+            JToken JId;
+            if(!firstData.TryGetValue("id", out JId))
+                isIncorrectRequest = true;
+
+            if (isIncorrectRequest)
+                return null;
+
+            //Convert JTokens to .NET types. 
+            int id = JId.ToObject<int>();
+            string password = jPassword.ToObject<string>();
+
+            //Get the camera with its id
+            Camera camera = FindCamera(id);
```

Comment:
Maybe `this.cameras.FirstOrDefault` with Linq instead of loop?

Reply:
Can be resolved I think?